src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -725,8 +725,7 @@ private void computeUpstreamNodesWhenWorkspaceModified(WorkspacesModificationEve
             if (args.Id == this.Guid)
             {
                 this.workspaceLoaded = true;
-                var frozenNodes = nodes.Where(x => x.isFrozenExplicitly).ToList();
-                frozenNodes.ForEach(z => z.ComputeUpstreamOnDownstreamNodes());
+                this.ComputeUpstreamCacheForEntireGraph();
```

Comment:
Also now, the UpstreamCache is correct - if other callers use this cache for something else.